src/service.rs

```diff
@@ -760,102 +641,87 @@ pub async fn run_non_permissioned_standalone_builder_service<
 /*
 Utility functions to handle the hotshot events
 */
+#[instrument(skip_all, fields(sender, da_proposal.data.view_number))]
 async fn handle_da_event<TYPES: NodeType>(
     da_channel_sender: &BroadcastSender<MessageType<TYPES>>,
     da_proposal: Proposal<TYPES, DaProposal<TYPES>>,
     sender: <TYPES as NodeType>::SignatureKey,
-    leader: <TYPES as NodeType>::SignatureKey,
-    total_nodes: NonZeroUsize,
 ) {
-    tracing::debug!(
-        "DaProposal: Leader: {:?} for the view: {:?}",
-        leader,
-        da_proposal.data.view_number
-    );
-
     // get the encoded transactions hash
     let encoded_txns_hash = Sha256::digest(&da_proposal.data.encoded_transactions);
     // check if the sender is the leader and the signature is valid; if yes, broadcast the DA proposal
-    if leader == sender && sender.validate(&da_proposal.signature, &encoded_txns_hash) {
-        let view_number = da_proposal.data.view_number;
-        tracing::debug!(
-            "Sending DA proposal to the builder states for view {:?}",
-            view_number
-        );
+    if !sender.validate(&da_proposal.signature, &encoded_txns_hash) {
```

Comment:
It wasn't necessary from the get go, because HotShot simply won't emit an event if the sender isn't the leader. Technically, it's true for the signature validation as well, but I left it be as a sanity check.